### DISCUSSION TITLE
Option to disable initial chain switching when connecting

### DISCUSSION THREAD

**Discussion by pedromcunha** [rainbow-me/rainbowkit#1734] [Ideas]
Our application waits until the user is ready to commit an action before switching them to the right chain, this results in better frictionless ux. There is however an issue if the user connects and is not already on the first chain or initialChain, RK will trigger the chain switching ux of the wallet (in this case metamask). Is there a way to disable this? Judging from the [code](https://github.com/rainbow-me/rainbowkit/blob/226050917f775231ec49ce0e9cc7406aaebab13a/packages/rainbowkit/src/wallets/useWalletConnectors.ts#L37) it doesn't seem like it's possible.

**Comment by 0xmkh** [comment]
@pedromcunha We could probably expose a prop where it'll ignore the chain id when connecting to a wallet. 

We did this to ensure that the wallet is connected to one of the right chains which you've provided to the dApp and also to ensure that there will be no unexpected behaviours. Will keep this discussion open and let you know if there's any updates 🙏

**Reply by pedromcunha** [reply]
Hey any updates here? Would be a nice ux improvement in our application. Thanks.

**Reply by 0xmkh** [reply]
Hey there. Yes working on this PR https://github.com/rainbow-me/rainbowkit/pull/1806, please bear with us 🙏
